src/v/cloud_storage/remote_segment.cc

```diff
@@ -828,9 +829,22 @@ ss::future<> remote_segment::run_hydrate_bg() {
         vlog(_ctxlog.error, "Error in hydraton loop: {}", err);
         set_waiter_errors(err);
     }
+
+    _hydration_loop_running = false;
 }
 
 ss::future<> remote_segment::hydrate() {
+    if (!_hydration_loop_running) {
+        vlog(
+          _ctxlog.error,
+          "Segment {} hydration requested, but the hydration loop is not "
+          "running",
+          _path);
+
+        return ss::make_exception_future<>(std::runtime_error(
+          fmt::format("Hydration loop is not running for segment: {}", _path)));
+    }
+
     gate_guard g{_gate};
```

Comment:
nit: callers of hydrate() already take the gate, but since we're already taking it here, it probably makes it easier to grok lifecycle moving it before the loop running check